pvanalytics/quality/irradiance.py

```diff
@@ -660,3 +660,271 @@ def calculate_component_sum_series(solar_zenith,
+    value > lower bound and value < upper bound. Also, steps with
+    change in magnitude of more than 1000 W/m2 are flagged. Lower bounds are
+    constant for all tests. Upper bounds are calculated as
+
+    .. math::
+        upper\_limit = 0.9 * dni\_extra * cos(aoi)^{1.2} + 300
+
+    Parameters
+    ----------
+    poa_global : Series
+        Global tilted irradiance in :math:`W/m^2`
+    solar_zenith : Series
+        Solar zenith angle in degrees
+    aoi : Series
+        Direct normal irradiance in :math:`W/m^2`
+    dni_extra : float
+        normal irradiance at the top of atmosphere in :math:`W/m^2`
+
+    Returns
+    -------
+    poa_global_limit_bool_flag : Series
+        True for each value that is physically possible.
+    poa_global_limit_int_flag : Series
+        Series of integers representing the flag numbers described in the
+        literature. [1]_
+
+    Notes
+    -----
+    The upper limit for `poa_global` is set to 0 when `solar_zenith` is greater
+    than 90 degrees. Missing values of `poa_global`, `solar_zenith`
+    and/or `aoi` will result in a `False` flag.
+
+    References
+    ----------
+    .. [1] Elke Lorenz et al., High resolution measurement network of global
+           horizontal and tilted solar irradiance in southern Germany with a
+           new quality control scheme, Solar Energy, Volume 231, 2022,
+           Pages 593-606, ISSN 0038-092X,
+           https://doi.org/10.1016/j.solener.2021.11.023.
+    """
+    # Finding the upper and lower limit
+    upper_limit = _upper_poa_global_limit_lorenz(aoi, solar_zenith, dni_extra)
+    lower_limit = _lower_limit_lorenz(solar_zenith, dni_extra)
+
+    # Initiating a poa_global_limit_int_flag series
+    poa_global_limit_int_flag = pd.Series(0, index=solar_zenith.index)
+
+    # Initiating a poa_global_limit_bool_flag series
+    poa_global_limit_bool_flag = pd.Series(True, index=solar_zenith.index)
+
+    # Changing the poa_global_flag to 3 when poa_global is above upper
+    # limit or below lower limit
+    poa_global_limit_int_flag = poa_global_limit_int_flag.mask(
+        ((poa_global > upper_limit) |
+         (poa_global < lower_limit)),
+        3
+    )
+
+    # Changing the poa_global_flag to 3 when the step change in poa values is
+    # more than 1000 W/m2
+    poa_global_limit_int_flag = poa_global_limit_int_flag.mask(
+        (abs(poa_global - poa_global.shift(1)) > 1000),
+        3
+    )
+
+    # Changing the poa_global_flag to 1 when poa_global is not available
+    poa_global_limit_int_flag = poa_global_limit_int_flag.mask(
+        ((poa_global.isna()) |
+         (upper_limit.isna()) |
+         (lower_limit.isna())),
+        1
+    )
+
+    # Changing the poa_global_limit_bool_flag depending on
+    # poa_global_limit_int_flag
+    poa_global_limit_bool_flag = poa_global_limit_bool_flag.mask(
+        cond=poa_global_limit_int_flag != 0,
+        other=False)
```

Comment:
```suggestion
    poa_global_limit_bool_flag = poa_global_limit_int_flag != 0
```

Does that do the same thing?  If so, can get rid of the initialization at the top as well

Reply:
Neat! A slight modification: 

`poa_global_limit_bool_flag = poa_global_limit_int_flag == 0`.

Will remove the initialization, and apply similar changes for GHI.